pkg/objstore/azure/helpers.go

```diff
@@ -19,6 +19,11 @@ const DirDelim = "/"
 
 var errorCodeRegex = regexp.MustCompile(`X-Ms-Error-Code:\D*\[(\w+)\]`)
 
+func init() {
+	// This needs to be done at startup because the underlying variable is not thread safe.
+	pipeline.SetForceLogEnabled(false)
```

Comment:
Would someone ever need to enable this?

Reply:
I agree it's a very peculiar design to have this on by default; however, [`ForceLog` enabled by default](https://github.com/Azure/azure-pipeline-go/blob/dc95902f1d32034f8f743ccc6c3f2eb36b84da27/pipeline/core.go#L266), even though they [recommend using `ForceLog` sparingly](https://github.com/Azure/azure-pipeline-go/blob/dc95902f1d32034f8f743ccc6c3f2eb36b84da27/pipeline/defaultlog.go#L4-L5). It seems the authors of the azure-storage-blob-go module disagree, because they enable force logging in most of their errors.

Reply:
Ah, I misunderstand your comment. Are you asking if this should be configurable?

Reply:
Yes, exactly the latter one. I'm ok to just disable it until someone actually asks for it.

Reply:
I think we should do this for now. I'm going to try and fix this upstream and, if that goes ok, we will probably be able to take this out when we upgrade to use the fixed code.